blueflood-http/src/integration-test/java/com/rackspacecloud/blueflood/inputs/handlers/HttpHandlerIntegrationTest.java

```diff
@@ -413,4 +427,66 @@ public static void shutdown() {
         esSetup.terminate();
         vendor.shutdown();
     }
+
+    //TODO: This class should go away when we implement queries for enums. All the logic contained in this class
+    // should likely be transferred to AstyanaxReader or a process in the query pipeline that processes enums.
+    public class FakeEnumAstyanaxReader extends AstyanaxReader {
+
+        public <T extends Rollup> Points<T> getDataToRoll(Class<T> type, Locator locator, Range range, ColumnFamily<Locator, Long> cf) throws IOException {
+            Points<T> points = super.getDataToRoll(type, locator, range, cf);
+            return getEnumStringValuesFromHashes(points, locator);
+        }
+
+        private <T extends Rollup> Points<T> getEnumStringValuesFromHashes(Points<T> points, Locator locator) {
+            ColumnList<Long> enumvalues = getColumnsFromEnumCF(locator);
+            Map<Long, String> hash2enumValues = getEnumValueFromHash(enumvalues);
+            Points<T> pointsEnum = new Points<T>();
+
+            Map<Long, Points.Point<T>> pointsMap = points.getPoints();
+
+            for (Long timestamp : points.getPoints().keySet()) {
+                BluefloodEnumRollup enumRollup = (BluefloodEnumRollup)pointsMap.get(timestamp).getData();
+                for (Long hash : enumRollup.getHashedEnumValuesWithCounts().keySet()) {
+                    String enumValueString = hash2enumValues.get(hash);
+                    enumRollup.getStringEnumValuesWithCounts().put(enumValueString, enumRollup.getHashedEnumValuesWithCounts().get(hash));
+                    pointsEnum.add(new Points.Point<T>(timestamp,(T)enumRollup));
+                }
+            }
+            return pointsEnum;
+        }
+
+        private Map<Long, String> getEnumValueFromHash(ColumnList<Long> enumValues) {
+            HashMap<Long,String> hash2enumValues = new HashMap<Long, String>();
+
+            for (Column<Long> col: enumValues) {
+                hash2enumValues.put(col.getName(), col.getStringValue());
+            }
+
+            return hash2enumValues;
+        }
+
+        private ColumnList<Long> getColumnsFromEnumCF(final Locator locator) {
+            final Map<Locator, ColumnList<Long>> columns = new HashMap<Locator, ColumnList<Long>>();
+
+            try {
+                OperationResult<Rows<Locator, Long>> query = getKeyspace()
+                        .prepareQuery(CassandraModel.CF_METRICS_ENUM)
+                        .getKeySlice(locator)
```

Comment:
NIT: It probably makes sense to use getKey() instead of getKeySlice() here, since you only want to get the single locator row back.  Maybe that's how we should do it when we move the class to AstyanaxReader.